lib/gateways/s3.rb

```diff
@@ -0,0 +1,29 @@
+module Gateways
+  class S3
```

Comment:
Yeah the name spacing is all off across the app. Gateways are namespaced, UseCases are not. I think this requires a separate PR. I also prefer the `Something::Something::Something` structure, rather than nesting on new lines, as it makes it easier to search for in the codebase, especially when projects become big.

Reply:
Agreed that this one is for another PR.